localstack/services/lambda_/invocation/lambda_service.py

```diff
@@ -91,7 +91,7 @@ def stop(self) -> None:
         shutdown_futures = []
         for event_manager in self.event_managers.values():
             shutdown_futures.append(self.task_executor.submit(event_manager.stop))
-        # TODO: switch shutdown order?
+        # TODO: switch shutdown order? yes, shutdown starting versions before the running versions would make more sense
         for version_manager in self.lambda_running_versions.values():
             shutdown_futures.append(self.task_executor.submit(version_manager.stop))
         for version_manager in self.lambda_starting_versions.values():
```

Comment:
We learned a lesson that the `qualified_arn` is not unique when dealing with updates of `$LATEST`.

Albeit unlikely, this could potentially be a problem as well in the `lambda_service` where we use the `qualified_arn` to manager starting and running versions as well as event managers. I guess concurrent updates on latest are prohibited (?), which would limit the issue.